### PR TITLE
Onboard clients only if they are on same version as provider

### PR DESCRIPTION
**Pull request by leelavg**
- Operator major and minor version of client and provider should match for onboarding process to begin
- After this commit older clients will fail to get onboarded

### CODE REVIEW

**Comment by leelavg** [comment]
Testing:

```
=== ClientOperatorVersion is not provider, older clients
# grpcurl -d '{"consumerName": "d1344bdd-ce33-4b55-a5b7-0ca41d8b8e46", "onboardingTicket": "eyJp[...]FRY="}' --insecure 127.1.27.1:50051 provider.OCSProvider.OnboardConsumer
ERROR:
  Code: InvalidArgument
  Message: malformed ClientPlatformVersion for client "d1344bdd-ce33-4b55-a5b7-0ca41d8b8e46" is provided. Version string empty

=== ClientOperatorVersion is empty, incorrect data
# grpcurl -d '{"consumerName": "d1344bdd-ce33-4b55-a5b7-0ca41d8b8e46", "clientOperatorVersion": "", "onboardingTicket": "eyJp[...]FRY="}' --insecure 127.1.27.1:50051 provider.OCSProvider.OnboardConsumer
ERROR:
  Code: InvalidArgument
  Message: malformed ClientPlatformVersion for client "d1344bdd-ce33-4b55-a5b7-0ca41d8b8e46" is provided. Version string empty

=== ClientOperatorVersion minor version mismatch, provider is running on 4.15.0
# grpcurl -d '{"consumerName": "d1344bdd-ce33-4b55-a5b7-0ca41d8b8e46", "clientOperatorVersion": "4.14.3", "onboardingTicket": "eyJp[...]FRY="}' --insecure 127.1.27.1:50051 provider.OCSProvider.OnboardConsumer
ERROR:
  Code: InvalidArgument
  Message: both major and minor version of server and client "d1344bdd-ce33-4b55-a5b7-0ca41d8b8e46" should match for onboarding process

=== Supported ODF Client
# grpcurl -d '{"consumerName": "d1344bdd-ce33-4b55-a5b7-0ca41d8b8e46", "clientOperatorVersion": "4.15.3", "onboardingTicket": "eyJp[...]FRY="}' --insecure 127.1.27.1:50051 provider.OCSProvider.OnboardConsumer
{
  "storageConsumerUUID": "e69e0e07-3de5-45ca-97e7-255c1d371b0e"
}
```

File: services/provider/server/server.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/v4/api/v1alpha1"
 	controllers "github.com/red-hat-storage/ocs-operator/v4/controllers/storageconsumer"
 	pb "github.com/red-hat-storage/ocs-operator/v4/services/provider/pb"
+	provVersion "github.com/red-hat-storage/ocs-operator/v4/version"
```
Comment (iamniting) [inline services/provider/server/server.go:24]:
```suggestion
	ocsVersion "github.com/red-hat-storage/ocs-operator/v4/version"
```

File: services/provider/server/server.go [hunk 2]
```diff
@@ -80,17 +82,32 @@ func NewOCSProviderServer(ctx context.Context, namespace string) (*OCSProviderSe
 		return nil, fmt.Errorf("failed to create new StorageClassRequest instance. %v", err)
 	}
 
+	version, err := semver.Make(provVersion.Version)
```
Comment (iamniting) [inline services/provider/server/server.go:85]:
```suggestion
	version, err := semver.Make(ocsVersion.Version)
```

**Comment by nb-ohad** [comment]
@leelavg This PR got merged before a reviewer with proper context around provider mode could review it. 

I can see 2 options here:
1. we will leave comments on this PR and you will address them in another PR
2. Have another PR based on Main~1 which will include the changes to this PR and any changes on top. In this case, we need to be very careful to do proper rebasing before merging. 

Please do the needed

FYI @iamniting

**Pull request merged**